pkg/provider/aws/plugin.go

```diff
@@ -55,6 +59,49 @@ func (a *AWS) Translate(result *core.ConstructGraph, dag *core.ResourceGraph) (L
 	return
 }
 
+// createEksCluster determines whether any execution units have a type of EKS to determine whether a cluster needs to be created.
+//
+// If any units do have a type of EKS, the function will look at their configuration to determine how the cluster needs to be configured.
+// The clusterId field within an execution units configuration, will determine which cluster the unit will belong to, helping klotho understands how many clusters to create.
+// If the clusterId field is unassigned, the execution unit will be assigned to the first clusterId alphanumerically.
+// If there are no clusterIds defined by any units, one cluster will be created for all units.
+func (a *AWS) createEksClusters(result *core.ConstructGraph, dag *core.ResourceGraph) {
+	unassignedUnits := []*core.ExecutionUnit{}
+	clusterIdToUnit := map[string][]*core.ExecutionUnit{}
+	for _, unit := range core.GetResourcesOfType[*core.ExecutionUnit](result) {
+		cfg := a.Config.GetExecutionUnit(unit.Provenance().ID)
+		if cfg.Type == Kubernetes {
+			params := cfg.GetExecutionUnitParamsAsKubernetes()
+			if params.ClusterId == "" {
+				unassignedUnits = append(unassignedUnits, unit)
+				continue
+			}
+			clusterIdToUnit[params.ClusterId] = append(clusterIdToUnit[params.ClusterId], unit)
+
+		}
+	}
+
+	//Assign unassigned units to the first key after sorted
+	keys := []string{}
```

Comment:
I'm not sure how I feel about assigning it to the first one — that feels really arbitrary. If there's only one key, assigning the units to it makes sense. But otherwise, it feels like we should just error out.

Reply:
hmmm yeah maybe thats the right method